### PR TITLE
fix: stuck snapshots

### LINKED ISSUE #199: Snapshot finaliser fails
```
2023-05-19T11:08:52Z	ERROR	Reconciler error	{"controller": "snapshot", "controllerGroup": "delivery.ocm.software", "controllerKind": "Snapshot", "Snapshot": {"name":"cache-localization-rkkqihh","namespace":"mpas-sample-project"}, "namespace": "mpas-sample-project", "name": "cache-localization-rkkqihh", "reconcileID": "c92f5197-a251-459c-8caf-94dc88c6c71d", "error": "failed to remove finalizer: failed to delete data: failed to fetch head for reference: HEAD http://registry.ocm-system.svc.cluster.local:5000/v2/sha-7579674839716172274/manifests/5423: unexpected status code 404 Not Found (HEAD responses have no body, use GET for details)"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/Users/skarlso/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/Users/skarlso/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/Users/skarlso/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235
```

### PR DESCRIPTION
**Pull request by Skarlso**
Closes #199

### CODE REVIEW

File: controllers/snapshot_controller.go [hunk 1]
```diff
@@ -152,7 +153,7 @@ func (r *SnapshotReconciler) reconcileDeleteSnapshot(ctx context.Context, obj *d
 
 	if err := r.Cache.DeleteData(ctx, name, obj.Spec.Tag); err != nil {
 		var terr *transport.Error
-		if !errors.As(err, &terr) || !containsManifestNotFoundError(terr.Errors) {
+		if !errors.As(err, &terr) || (terr.StatusCode != http.StatusNotFound && !containsManifestNotFoundError(terr.Errors)) {
```
Comment (phoban01) [inline controllers/snapshot_controller.go:156]: this is very confusing to read 😄 ; could we refactor slightly for readbility?

**Review by phoban01 — COMMENTED** [review]
Great to get this fixed!

File: controllers/snapshot_controller.go (hunk 1 above)
Comment (Skarlso) [reply controllers/snapshot_controller.go:156]: :D :D sure. :)

Comment (Skarlso) [reply controllers/snapshot_controller.go]: Amended. :))) 👍

**Review by phoban01 — APPROVED** [review]

**Pull request merged**